server/test/api/depots.js

```diff
@@ -0,0 +1,108 @@
+/* jshint expr: true */
+/* global describe, it, beforeEach */
+
+var chai = require('chai');
+var expect = chai.expect;
+
+/** import test helpers */
+var helpers = require('./helpers');
+var uuid    = require('node-uuid');
+helpers.configure(chai);
+
+
+/**
+* The /depots API endpoint
+*
+* @desc This test suit is about the crud operation with depots
+*
+*/
+describe('The /depots API endpoint :: ', function () {
+  'use strict';
+
+  var agent = chai.request.agent(helpers.baseUrl);
+
+  /** login before each request */
+  beforeEach(helpers.login(agent));
+
+  /** new depot object */
+  var newDepot = {
+    uuid : uuid.v4(),
+    /** the reference column is auto increment */
+    text : 'New Depot',
+    enterprise_id : 1,
+    is_warehouse : 0
+  };
+
+  /** debot object with missing uuid */
+  var badDepot = {
+    text : 'New Depot',
+    enterprise_id : 1,
+    is_warehouse : 0
+  };
+
+  /** update depot */
+  var editDepot = {
+    uuid : uuid.v4(),
+    text : 'Edited Depot',
+    is_warehouse : 1
+  };
+
+  it('POST /depots create a new depot in the database', function () {
+    return agent.post('/depots')
+    .send(newDepot)
+    .then(function (res) {
+      expect(res).to.have.status(201);
+      expect(res.body.id).to.exist;
+      expect(res.body.id).to.be.equal(newDepot.uuid);
+    })
+    .catch(helpers.handler);
+  });
+
+  it('POST /depots should not create when missing uuid', function () {
+    return agent.post('/depots')
+    .send(badDepot)
+    .then(function (res) {
+      expect(res).to.have.status(400);
+      expect(res.body.code).to.exist;
+      expect(res.body.code).to.be.equal('DB.ER_NO_DEFAULT_FOR_FIELD');
+    })
+    .catch(helpers.handler);
+  });
+
+  it('POST /depots should not create when missing data', function () {
+    badDepot.uuid = uuid.v4();
+    delete badDepot.enterprise_id;
+
+    return agent.post('/depots')
+    .send(badDepot)
+    .then(function (res) {
+      expect(res).to.have.status(400);
+      expect(res.body.code).to.exist;
+      expect(res.body.code).to.be.equal('DB.ER_NO_DEFAULT_FOR_FIELD');
+    })
+    .catch(helpers.handler);
+  });
+
+  it('GET /depots should returns the list of depots', function () {
```

Comment:
How many depots do you expect to receive?  That would make a much stronger test.